server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/StarTreeQueryHelper.java

```diff
@@ -74,10 +79,16 @@ public static StarTreeQueryContext getStarTreeQueryContext(SearchContext context
         );
 
         for (AggregatorFactory aggregatorFactory : context.aggregations().factories().getFactories()) {
-            MetricStat metricStat = validateStarTreeMetricSupport(compositeMappedFieldType, aggregatorFactory);
-            if (metricStat == null) {
-                return null;
+            // first check for aggregation is a metric aggregation
+            if (validateStarTreeMetricSupport(compositeMappedFieldType, aggregatorFactory)) {
+                continue;
+            }
+
+            // if not a metric aggregation, check for applicable date histogram shape
+            if (validateDateHistogramSupport(compositeMappedFieldType, aggregatorFactory)) {
+                continue;
             }
```

Comment:
As we support more aggregations, I don't think this approach will scale. 

For now, though, I don't think it's a blocker.